transaction-status/src/parse_token/extension/token_metadata.rs

```diff
@@ -0,0 +1,347 @@
+            symbol.clone(),
+            uri.clone(),
+        );
+        let mut message = Message::new(&[ix], None);
+        let compiled_instruction = &mut message.instructions[0];
+        assert_eq!(
+            parse_token(
+                compiled_instruction,
+                &AccountKeys::new(&message.account_keys, None)
+            )
+            .unwrap(),
+            ParsedInstructionEnum {
+                instruction_type: "initializeTokenMetadata".to_string(),
+                info: json!({
+                    "metadata": metadata.to_string(),
+                    "updateAuthority": update_authority.to_string(),
+                    "mint": mint.to_string(),
+                    "mintAuthority": mint_authority.to_string(),
+                    "name": name,
+                    "symbol": symbol,
+                    "uri": uri,
+                })
+            }
+        );
+
+        // UpdateField
+        // Update one of the fixed fields.
+        let ix = spl_token_metadata_interface::instruction::update_field(
+            &spl_token_2022::id(),
+            &metadata,
+            &update_authority,
+            spl_token_metadata_interface::state::Field::Uri,
+            "https://ultra-mega.com".to_string(),
+        );
+        let mut message = Message::new(&[ix], None);
+        let compiled_instruction = &mut message.instructions[0];
+        assert_eq!(
+            parse_token(
+                compiled_instruction,
+                &AccountKeys::new(&message.account_keys, None)
+            )
+            .unwrap(),
+            ParsedInstructionEnum {
+                instruction_type: "updateTokenMetadataField".to_string(),
+                info: json!({
+                    "metadata": metadata.to_string(),
+                    "updateAuthority": update_authority.to_string(),
+                    "field": "uri",
+                    "value": "https://ultra-mega.com",
+                })
+            }
+        );
+        // Add a new field
+        let ix = spl_token_metadata_interface::instruction::update_field(
+            &spl_token_2022::id(),
+            &metadata,
+            &update_authority,
+            spl_token_metadata_interface::state::Field::Key("new_field".to_string()),
+            "new_value".to_string(),
+        );
+        let mut message = Message::new(&[ix], None);
+        let compiled_instruction = &mut message.instructions[0];
+        assert_eq!(
+            parse_token(
+                compiled_instruction,
+                &AccountKeys::new(&message.account_keys, None)
+            )
+            .unwrap(),
+            ParsedInstructionEnum {
+                instruction_type: "updateTokenMetadataField".to_string(),
+                info: json!({
+                    "metadata": metadata.to_string(),
+                    "updateAuthority": update_authority.to_string(),
+                    "field": {
+                        "key": "new_field",
+                    },
```

Comment:
This is a bit gross... What do you think about flattening this to `"field": "new_field"`? there is the off-chance of someone adding a custom "name" field and getting confused, but I would prefer for them to be confused than everyone else to have to deal with this nested object.

Reply:
I thought about doing this, but since we already implement `serde` on the type I figured use the already-implemented trait. But if you think this is better, and it's only the one type, we can do it!